{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d65e4176",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ──────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.3     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.4.3     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.2     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.2     \n",
      "── \u001b[1mConflicts\u001b[22m ────────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:lubridate’:\n",
      "\n",
      "    hour, isoweek, mday, minute, month, quarter, second, wday, week,\n",
      "    yday, year\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(data.table)\n",
    "library(readxl)\n",
    "library(dplyr)\n",
    "library(stringr)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c2ee87f",
   "metadata": {},
   "source": [
    "Make a bedfile of the non-suspicious indels where the second position is length of ref - 1. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "eb04c661",
   "metadata": {},
   "outputs": [],
   "source": [
    "## This script is to get a bedfile for rAF-lo IGM indels \n",
    "\n",
    "## The columns for the bedfiles are: CHR | START | END\n",
    "\n",
    "setwd (\"IGM/\")\n",
    "\n",
    "rAFlo_bp10 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp10_rAF_loIndels.lt50bp.csv\")\n",
    "rAFlo_bp20 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp20_rAF_loIndels.lt50bp.csv\")\n",
    "rAFlo_bp30 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp30_rAF_loIndels.lt50bp.csv\")\n",
    "rAFlo_bp40 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp40_rAF_loIndels.lt50bp.csv\")\n",
    "\n",
    "df_name = \"IGM\"\n",
    "\n",
    "\n",
    "bp_range = c(\"10\", \"20\", \"30\", \"40\")\n",
    "\n",
    "for (i in bp_range) {\n",
    "  get_df = paste0(\"rAFlo_bp\", i)\n",
    "  \n",
    "  bed = paste0(\"bed_bp\", i)\n",
    "  \n",
    "  assign(get_df, get(get_df) %>% separate(VarID, c(\"CHR\", \"POS\", \"REF\", \"ALT\")))\n",
    "  assign(get_df, get(get_df) %>% mutate(refLen = nchar(get(get_df)$REF)))\n",
    "  assign(bed, data.frame(get(get_df)$CHR, get(get_df)$POS, as.numeric(get(get_df)$POS) + as.numeric(get(get_df)$refLen) - 1))\n",
    "\n",
    "  fwrite(get(bed), paste0(df_name, \"_rAF_lo_indels_bp\", i, \".bed\"), col.names=FALSE, sep = \"\\t\")\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a9da533",
   "metadata": {},
   "source": [
    "Merge bedfile of gnomAD bins that are sAF-hi and rAF-hi. \n",
    "Bash script below. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9ff935af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'10'</li><li>'20'</li><li>'30'</li><li>'40'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '10'\n",
       "\\item '20'\n",
       "\\item '30'\n",
       "\\item '40'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '10'\n",
       "2. '20'\n",
       "3. '30'\n",
       "4. '40'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"10\" \"20\" \"30\" \"40\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system(\"bp_range=('10' '20' '30' '40')\n",
    "\n",
    "date='2023-09-12'\n",
    "output_path='/gnomAD/'\n",
    "\n",
    "df_name='gnomAD'\n",
    "\n",
    "for i in ${bp_range[@]};\n",
    "do\n",
    "   echo $i\n",
    "   cat ${output_path}gnomad.exomes.r2.1.1.sites_indelsonly_rAF_bp${i}_rAF_hiIndels.lt50bp.region.bed ${output_path}gnomad.exomes.r2.1.1.sites_indelsonly_rAF_bp${i}_sAF_hiIndels.lt50bp.region.bed > ${output_path}${df_name}_rAFhi_and_sAFhi_bins_bp${i}.bed\n",
    "done;\", intern = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42d969bd",
   "metadata": {},
   "source": [
    "Bedtools intersect IGM indels that are rAF-lo by IGM indel with gnomAD bins that are sAF-hi or rAF-hi. This would combine the regions that are also sAF-hi or rAF-hi by gnomAD.\n",
    "\n",
    "Bash script below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5719ecf5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'10'</li><li>'20'</li><li>'30'</li><li>'40'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '10'\n",
       "\\item '20'\n",
       "\\item '30'\n",
       "\\item '40'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '10'\n",
       "2. '20'\n",
       "3. '30'\n",
       "4. '40'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"10\" \"20\" \"30\" \"40\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system(\"bp_range=('10' '20' '30' '40')\n",
    "\n",
    "date='2023-09-12'\n",
    "output_path_df2='/IGM/'\n",
    "output_path_df1='/gnomAD/'\n",
    "\n",
    "\n",
    "df_1_name='gnomAD'\n",
    "df_2_name='IGM'\n",
    "\n",
    "for i in ${bp_range[@]};\n",
    "do\n",
    "   echo $i\n",
    "   bedtools intersect -wa -a ${output_path_df2}${df_2_name}_rAF_lo_indels_bp${i}.bed -b ${output_path_df1}${df_1_name}_rAFhi_and_sAFhi_bins_bp${i}.bed | sort -k1,1n -k2,2n | uniq > ${output_path_df2}${df_2_name}_rAF_lo_indels_annotated_bp${i}.bed\n",
    "done;\", intern = TRUE)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed1401b1",
   "metadata": {},
   "source": [
    "Get files with: \n",
    " 1. File 1 (all_sus): [indels that are sus by IGM rAF] + [indels that are overlapping with gnomAD sus and common bins]\n",
    " 2. File 2 (reduced_sus): [indels that are non-sus by IGM rAF] - [indels that are overlapping with gnomAD sus and common bins]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0c0a051b",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(data.table)\n",
    "library(readxl)\n",
    "library(dplyr)\n",
    "library(stringr)\n",
    "\n",
    "setwd (\"/IGM\")\n",
    "\n",
    "\n",
    "## This script is to get a file with all the rAFhi IGM indels, including the indels that are rAFlo by IGM rAF but are common/rAFhi by gnomAD rAF \n",
    "\n",
    "bp_range = c(\"10\", \"20\", \"30\", \"40\")\n",
    "output_path = \"\"\n",
    "\n",
    "df_name = \"IGM\"\n",
    "\n",
    "###############################################################\n",
    "###############################################################\n",
    "\n",
    "## Read in IGM indels that are rAFhi by IGM rAF only \n",
    "rAF_hi_bp10 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp10_rAF_hiIndels.lt50bp.csv\")\n",
    "rAF_hi_bp20 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp20_rAF_hiIndels.lt50bp.csv\")\n",
    "rAF_hi_bp30 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp30_rAF_hiIndels.lt50bp.csv\")\n",
    "rAF_hi_bp40 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp40_rAF_hiIndels.lt50bp.csv\")\n",
    "\n",
    "## Read in rAFlo IGM indels \n",
    "rAF_lo_bp10 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp10_rAF_loIndels.lt50bp.csv\")\n",
    "rAF_lo_bp20 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp20_rAF_loIndels.lt50bp.csv\")\n",
    "rAF_lo_bp30 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp30_rAF_loIndels.lt50bp.csv\")\n",
    "rAF_lo_bp40 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp40_rAF_loIndels.lt50bp.csv\")\n",
    "\n",
    "## Read in rAFlo IGM indels bedfile that are annotated to be non-rAFlo by gnomAD rAF and sAF\n",
    "annotated_bp10 = fread(\"IGM_rAF_lo_indels_annotated_bp10.bed\")\n",
    "annotated_bp20 = fread(\"IGM_rAF_lo_indels_annotated_bp20.bed\")\n",
    "annotated_bp30 = fread(\"IGM_rAF_lo_indels_annotated_bp30.bed\")\n",
    "annotated_bp40 = fread(\"IGM_rAF_lo_indels_annotated_bp40.bed\")\n",
    "\n",
    "###############################################################\n",
    "###############################################################\n",
    "\n",
    "## Map the annotated_bp indels to the rAF_lo_bp dataframes by CHR and first POS\n",
    "\n",
    "for (i in bp_range) {\n",
    "  rAF_lo_bp = paste0(\"rAF_lo_bp\", i)\n",
    "  annotated_bp = paste0(\"annotated_bp\", i)\n",
    "  \n",
    "  varID_annotated_bp = paste0(\"varID_annotated_bp\", i)\n",
    "  \n",
    "  assign(rAF_lo_bp, get(rAF_lo_bp) %>% separate(VarID, c(\"CHR\", \"POS\", \"REF\", \"ALT\")))\n",
    "  \n",
    "  assign(varID_annotated_bp, get(rAF_lo_bp) %>% filter(paste0(get(rAF_lo_bp)$CHR, \"-\", get(rAF_lo_bp)$POS) %in% paste0(get(annotated_bp)$V1, \"-\", get(annotated_bp)$V2)))\n",
    "}\n",
    "\n",
    "## Combine the rAFhi indels that are rAFhi by IGM rAF with the annotated rAFlo indels that are rAFhi by gnomAD rAF \n",
    "\n",
    "for (i in bp_range) {\n",
    "  varID_annotated_bp = paste0(\"varID_annotated_bp\", i)\n",
    "  assign(varID_annotated_bp, subset((get(varID_annotated_bp) %>% mutate(VarID = paste0(CHR, \"-\", POS, \"-\", REF, \"-\", ALT))), select = -c(CHR, POS, REF, ALT)))\n",
    "  \n",
    "  ## bind to rAFhi indels by IGM rAF\n",
    "  rAF_hi_bp = paste0(\"rAF_hi_bp\", i)\n",
    "  \n",
    "  ## all rAFhi (including annotated one)\n",
    "  all_rAF_hi_bp = paste0(\"all_rAF_hi_bp\", i)\n",
    "  \n",
    "  assign(all_rAF_hi_bp, rbind(get(varID_annotated_bp), get(rAF_hi_bp)))\n",
    "  fwrite(get(all_rAF_hi_bp), paste0(output_path, df_name, \"_all_rAF_hi_bp\", i, \".csv\"))\n",
    "}\n",
    "\n",
    "## Find the reduced rAF_lo indels.\n",
    "## rAFlo indels that are NOT in gnomAD bins that are common/rAF_hi. \n",
    "for (i in bp_range) {\n",
    "  rAF_lo_bp = paste0(\"rAF_lo_bp\", i)\n",
    "  assign(rAF_lo_bp, subset((get(rAF_lo_bp) %>% mutate(VarID = paste0(CHR, \"-\", POS, \"-\", REF, \"-\", ALT))), select = -c(CHR, POS, REF, ALT)))\n",
    "  \n",
    "  varID_annotated_bp = paste0(\"varID_annotated_bp\", i)\n",
    "  reduced_rAF_lo_bp = paste0(\"reduced_rAF_lo_bp\", i)\n",
    "  \n",
    "  # exclude VarIDs that are in varID_annotated_bp from the rAF_lo_bp df \n",
    "  assign(reduced_rAF_lo_bp, get(rAF_lo_bp) %>% filter(!(VarID %in% get(varID_annotated_bp)$VarID)))\n",
    "  fwrite(get(reduced_rAF_lo_bp), paste0(output_path, df_name, \"_reduced_rAF_lo_bp\", i, \".csv\"))\n",
    "  \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "402c26af",
   "metadata": {},
   "source": [
    "Find the number of individuals that contain only suspicious deleterious indels by not counting the individuals that are also in the cohort with at least one suspicious deleterious indel in a constrained gene associated with an autosomal dominant disorder. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "85960a5a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>bp_range</th><th scope=col>rAF_lo_individuals</th><th scope=col>only_rAF_hi_individuals</th><th scope=col>prct_rAF_lo_out_of_total</th><th scope=col>prct_only_rAF_hi_out_of_total</th><th scope=col>prct_only_rAF_hi_out_of_rAF_lo</th><th scope=col>prct_only_rAF_hi_out_of_5299</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>10</td><td>4063</td><td>1236</td><td>10.32</td><td>3.14</td><td>30.42</td><td>23.33</td></tr>\n",
       "\t<tr><td>20</td><td>3790</td><td>1509</td><td> 9.63</td><td>3.83</td><td>39.82</td><td>28.48</td></tr>\n",
       "\t<tr><td>30</td><td>3569</td><td>1730</td><td> 9.07</td><td>4.39</td><td>48.47</td><td>32.65</td></tr>\n",
       "\t<tr><td>40</td><td>3373</td><td>1926</td><td> 8.57</td><td>4.89</td><td>57.10</td><td>36.35</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " bp\\_range & rAF\\_lo\\_individuals & only\\_rAF\\_hi\\_individuals & prct\\_rAF\\_lo\\_out\\_of\\_total & prct\\_only\\_rAF\\_hi\\_out\\_of\\_total & prct\\_only\\_rAF\\_hi\\_out\\_of\\_rAF\\_lo & prct\\_only\\_rAF\\_hi\\_out\\_of\\_5299\\\\\n",
       " <chr> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 10 & 4063 & 1236 & 10.32 & 3.14 & 30.42 & 23.33\\\\\n",
       "\t 20 & 3790 & 1509 &  9.63 & 3.83 & 39.82 & 28.48\\\\\n",
       "\t 30 & 3569 & 1730 &  9.07 & 4.39 & 48.47 & 32.65\\\\\n",
       "\t 40 & 3373 & 1926 &  8.57 & 4.89 & 57.10 & 36.35\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4 × 7\n",
       "\n",
       "| bp_range &lt;chr&gt; | rAF_lo_individuals &lt;int&gt; | only_rAF_hi_individuals &lt;int&gt; | prct_rAF_lo_out_of_total &lt;dbl&gt; | prct_only_rAF_hi_out_of_total &lt;dbl&gt; | prct_only_rAF_hi_out_of_rAF_lo &lt;dbl&gt; | prct_only_rAF_hi_out_of_5299 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 10 | 4063 | 1236 | 10.32 | 3.14 | 30.42 | 23.33 |\n",
       "| 20 | 3790 | 1509 |  9.63 | 3.83 | 39.82 | 28.48 |\n",
       "| 30 | 3569 | 1730 |  9.07 | 4.39 | 48.47 | 32.65 |\n",
       "| 40 | 3373 | 1926 |  8.57 | 4.89 | 57.10 | 36.35 |\n",
       "\n"
      ],
      "text/plain": [
       "  bp_range rAF_lo_individuals only_rAF_hi_individuals prct_rAF_lo_out_of_total\n",
       "1 10       4063               1236                    10.32                   \n",
       "2 20       3790               1509                     9.63                   \n",
       "3 30       3569               1730                     9.07                   \n",
       "4 40       3373               1926                     8.57                   \n",
       "  prct_only_rAF_hi_out_of_total prct_only_rAF_hi_out_of_rAF_lo\n",
       "1 3.14                          30.42                         \n",
       "2 3.83                          39.82                         \n",
       "3 4.39                          48.47                         \n",
       "4 4.89                          57.10                         \n",
       "  prct_only_rAF_hi_out_of_5299\n",
       "1 23.33                       \n",
       "2 28.48                       \n",
       "3 32.65                       \n",
       "4 36.35                       "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## Table S10.\n",
    "\n",
    "setwd (\"/IGM\")\n",
    "\n",
    "library(tidyverse)\n",
    "library(data.table)\n",
    "library(readxl)\n",
    "library(dplyr)\n",
    "library(stringr)\n",
    "\n",
    "bp_range = c(\"10\", \"20\", \"30\", \"40\")\n",
    "\n",
    "total_samples = 39367\n",
    "\n",
    "variants_effects = c(\"frameshift_variant\", \"splice_donor_variant\", \"splice_acceptor_variant\", \"stop_gained\", \"start_lost\", \"stop_lost\", \"exon_loss_variant\")\n",
    "\n",
    "## This script finds the number of individuals with only deleterious rAF_hi indels in constrained, autosomal dominant genes  \n",
    "all_rAF_hi_bp10 = fread(\"IGM_all_rAF_hi_bp10.csv\")\n",
    "all_rAF_hi_bp20 = fread(\"IGM_all_rAF_hi_bp20.csv\")\n",
    "all_rAF_hi_bp30 = fread(\"IGM_all_rAF_hi_bp30.csv\")\n",
    "all_rAF_hi_bp40 = fread(\"IGM_all_rAF_hi_bp40.csv\")\n",
    "\n",
    "## Read in rAF_lo IGM indels (removed the indels that are in common/rAF_hi gnomAD bins)\n",
    "reduced_rAF_lo_bp10 = fread(\"IGM_reduced_rAF_lo_bp10.csv\")\n",
    "reduced_rAF_lo_bp20 = fread(\"IGM_reduced_rAF_lo_bp20.csv\")\n",
    "reduced_rAF_lo_bp30 = fread(\"IGM_reduced_rAF_lo_bp30.csv\")\n",
    "reduced_rAF_lo_bp40 = fread(\"IGM_reduced_rAF_lo_bp40.csv\")\n",
    "\n",
    "sample_name_gene_name = fread(\"2023-03-24_11-47-14_IGM_n39367_indels_genotypes_selectcols.csv\") \n",
    "colnames(sample_name_gene_name) = c(\"VarID\", \"geneName\", \"sampleName\", \"coveredCtrl\", \"AC\")\n",
    "\n",
    "\n",
    "effects = fread(\"2023-04-21_IGM_n39367_indels_genotypes_effects.csv\", header = TRUE)\n",
    "colnames(effects) = c(\"VarID\", \"Effect\")\n",
    "\n",
    "annotations = distinct(fread (\"2023-04-21_IGM_genename_gnomadpli_gnomadloeuf_omimdisease.csv\")) #18231 \n",
    "colnames(annotations) = c(\"geneName\", \"pLI\", \"oe_lof_upper\", \"OMIM_disease\")\n",
    "\n",
    "annotations[geneName == \"'HTT'\"]$OMIM_disease = \"Huntington disease, 143100 (3), Autosomal dominant\"\n",
    "annotations[geneName == \"'GLTSCR1'\"]$geneName = \"BICRA\"\n",
    "annotations[geneName == \"'FAM46A'\"]$geneName = \"TENT5A\"\n",
    "\n",
    "\n",
    "## Merge IGM all rAF_hi indels with gene name, sample name, pLI, loeuf, omim, effects\n",
    "\n",
    "for (i in bp_range) {\n",
    "  all_rAF_hi_bp = paste0(\"all_rAF_hi_bp\", i)\n",
    "  \n",
    "  rAF_hi_with_gene_bp = paste0(\"rAF_hi_with_gene_bp\", i)\n",
    "  rAF_hi_with_effects_bp = paste0(\"rAF_hi_with_effects_bp\", i)\n",
    "  rAF_hi_with_annot_bp = paste0(\"rAF_hi_with_annot_bp\", i)\n",
    "  \n",
    "  assign (rAF_hi_with_gene_bp, merge(get(all_rAF_hi_bp), sample_name_gene_name, by=\"VarID\" ))\n",
    "  assign (rAF_hi_with_effects_bp, merge(get(rAF_hi_with_gene_bp), effects, by=\"VarID\"))\n",
    "  assign (rAF_hi_with_annot_bp, merge(get(rAF_hi_with_effects_bp), annotations, by=\"geneName\"))\n",
    "  ## rAF_hi dataframe with all the annotations is rAF_hi_with_annot_bp\n",
    "  \n",
    "  ## Now only look for rAF_hi indels of interest by filtering for indels that are autosomal dominant, deleterious, and constrained (pLI > 0.5; oe_lof_upper < 0.35)\n",
    "  filtered_rAF_hi_bp = paste0(\"filtered_rAF_hi_bp\", i)\n",
    "  assign(filtered_rAF_hi_bp, get(rAF_hi_with_annot_bp)[OMIM_disease %like% \"Autosomal dominant\", ] %>% filter(Effect %in% variants_effects) %>% filter(oe_lof_upper < 0.35) %>% filter(pLI > 0.5))\n",
    "  \n",
    "  ### Now look for rAF_lo indels \n",
    "  \n",
    "  reduced_rAF_lo_bp = paste0(\"reduced_rAF_lo_bp\", i)\n",
    "  \n",
    "  rAF_lo_with_gene_bp = paste0(\"rAF_lo_with_gene_bp\", i)\n",
    "  rAF_lo_with_effects_bp = paste0(\"rAF_lo_with_effects_bp\", i)\n",
    "  rAF_lo_with_annot_bp = paste0(\"rAF_lo_with_annot_bp\", i)\n",
    "  ## reduced rAF_lo dataframe with all the annotations is rAF_lo_with_annot_bp\n",
    "  \n",
    "  assign(rAF_lo_with_gene_bp, merge(get(reduced_rAF_lo_bp), sample_name_gene_name, by = \"VarID\"))\n",
    "  assign(rAF_lo_with_effects_bp, merge(get(rAF_lo_with_gene_bp), effects, by = \"VarID\"))\n",
    "  assign(rAF_lo_with_annot_bp, merge(get(rAF_lo_with_effects_bp), annotations, by =\"geneName\"))\n",
    "  \n",
    "  ## Now only look for non rAF_hi indels of interest by filtering for indels that are autosomal dominant, deleterious, and constrained (pLI > 0.5; oe_lof_upper < 0.35)\n",
    "  filtered_rAF_lo_bp = paste0(\"filtered_rAF_lo_bp\", i)\n",
    "  assign(filtered_rAF_lo_bp, get(rAF_lo_with_annot_bp)[OMIM_disease %like% \"Autosomal dominant\", ] %>% filter(Effect %in% variants_effects) %>% filter(oe_lof_upper < 0.35) %>% filter(pLI > 0.5))\n",
    "  \n",
    "  ## find unique individuals that are in filtered_rAF_lo_bp \n",
    "  rAF_lo_individuals = paste0(\"only_rAF_hi_individuals_bp\", i)\n",
    "  num_rAF_lo_individuals= paste0(\"num_rAF_lo_individuals_bp\", i)\n",
    "  \n",
    "  assign(rAF_lo_individuals, as.data.frame(unique(get(filtered_rAF_lo_bp)$sampleName)))\n",
    "  assign(num_rAF_lo_individuals, nrow(get(rAF_lo_individuals)))\n",
    "  \n",
    "  ## find unique rAF_hi individuals that are in filtered_rAF_lo_bp \n",
    "  ## this means rAF_hi indels that are NOT in non_rAF_hi indels \n",
    "  only_rAF_hi_individuals = paste0(\"only_rAF_hi_individuals_bp\", i )\n",
    "  num_only_rAF_hi_individuals = paste0(\"num_only_rAF_hi_individuals_bp\", i)\n",
    "  \n",
    "  assign(only_rAF_hi_individuals, as.data.frame(unique((get(filtered_rAF_hi_bp) %>% filter(!(get(filtered_rAF_hi_bp)$sampleName %in% get(filtered_rAF_lo_bp)$sampleName)))$sampleName)) )\n",
    "  assign(num_only_rAF_hi_individuals, nrow(get(only_rAF_hi_individuals)))\n",
    "}\n",
    "\n",
    "## make dataframe to visualize data \n",
    "rAF_lo_individuals = c(num_rAF_lo_individuals_bp10, num_rAF_lo_individuals_bp20, num_rAF_lo_individuals_bp30, num_rAF_lo_individuals_bp40)\n",
    "only_rAF_hi_individuals = c(num_only_rAF_hi_individuals_bp10, num_only_rAF_hi_individuals_bp20, num_only_rAF_hi_individuals_bp30, num_only_rAF_hi_individuals_bp40)\n",
    "  \n",
    "summary = data.frame(bp_range, rAF_lo_individuals, only_rAF_hi_individuals)\n",
    "\n",
    "## Find proportion out of all IGM cohort \n",
    "summary$prct_rAF_lo_out_of_total = round((as.numeric(summary$rAF_lo_individuals) / total_samples) * 100, 2) \n",
    "summary$prct_only_rAF_hi_out_of_total = round((as.numeric(summary$only_rAF_hi_individuals) / total_samples) * 100, 2) \n",
    "summary$prct_only_rAF_hi_out_of_rAF_lo = round((as.numeric(summary$only_rAF_hi_individuals) / (as.numeric(summary$rAF_lo_individuals))) * 100, 2) \n",
    "summary$prct_only_rAF_hi_out_of_5299 = round((as.numeric(summary$only_rAF_hi_individuals) / (5299)) * 100, 2) \n",
    "\n",
    "\n",
    "summary \n",
    "\n",
    "setwd(\"\")\n",
    "fwrite(summary, \"tableS10.csv\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bb737478",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
