{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "id": "d65e4176",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "── \u001b[1mAttaching core tidyverse packages\u001b[22m ────────────────────── tidyverse 2.0.0 ──\n",
      "\u001b[32m✔\u001b[39m \u001b[34mdplyr    \u001b[39m 1.1.1     \u001b[32m✔\u001b[39m \u001b[34mreadr    \u001b[39m 2.1.4\n",
      "\u001b[32m✔\u001b[39m \u001b[34mforcats  \u001b[39m 1.0.0     \u001b[32m✔\u001b[39m \u001b[34mstringr  \u001b[39m 1.5.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mggplot2  \u001b[39m 3.4.1     \u001b[32m✔\u001b[39m \u001b[34mtibble   \u001b[39m 3.2.1\n",
      "\u001b[32m✔\u001b[39m \u001b[34mlubridate\u001b[39m 1.9.2     \u001b[32m✔\u001b[39m \u001b[34mtidyr    \u001b[39m 1.3.0\n",
      "\u001b[32m✔\u001b[39m \u001b[34mpurrr    \u001b[39m 1.0.1     \n",
      "── \u001b[1mConflicts\u001b[22m ──────────────────────────────────────── tidyverse_conflicts() ──\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mfilter()\u001b[39m masks \u001b[34mstats\u001b[39m::filter()\n",
      "\u001b[31m✖\u001b[39m \u001b[34mdplyr\u001b[39m::\u001b[32mlag()\u001b[39m    masks \u001b[34mstats\u001b[39m::lag()\n",
      "\u001b[36mℹ\u001b[39m Use the conflicted package (\u001b[3m\u001b[34m<http://conflicted.r-lib.org/>\u001b[39m\u001b[23m) to force all conflicts to become errors\n",
      "\n",
      "Attaching package: ‘data.table’\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:lubridate’:\n",
      "\n",
      "    hour, isoweek, mday, minute, month, quarter, second, wday, week,\n",
      "    yday, year\n",
      "\n",
      "\n",
      "The following objects are masked from ‘package:dplyr’:\n",
      "\n",
      "    between, first, last\n",
      "\n",
      "\n",
      "The following object is masked from ‘package:purrr’:\n",
      "\n",
      "    transpose\n",
      "\n",
      "\n"
     ]
    }
   ],
   "source": [
    "library(tidyverse)\n",
    "library(data.table)\n",
    "library(readxl)\n",
    "library(dplyr)\n",
    "library(stringr)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "5c2ee87f",
   "metadata": {},
   "source": [
    "Make a bedfile of the non-suspicious indels where the second position is length of ref - 1. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "eb04c661",
   "metadata": {},
   "outputs": [],
   "source": [
    "## This script is to get a bedfile for non-suspicious IGM indels \n",
    "\n",
    "## The columns for the bedfiles are: CHR | START | END\n",
    "\n",
    "nonsus_bp10 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp10_RareIndels.lt50bp.csv\")\n",
    "nonsus_bp20 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp20_RareIndels.lt50bp.csv\")\n",
    "nonsus_bp30 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp30_RareIndels.lt50bp.csv\")\n",
    "nonsus_bp40 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp40_RareIndels.lt50bp.csv\")\n",
    "\n",
    "df_name = \"IGM\"\n",
    "\n",
    "\n",
    "bp_range = c(\"10\", \"20\", \"30\", \"40\")\n",
    "\n",
    "for (i in bp_range) {\n",
    "  get_df = paste0(\"nonsus_bp\", i)\n",
    "  \n",
    "  bed = paste0(\"bed_bp\", i)\n",
    "  \n",
    "  assign(get_df, get(get_df) %>% separate(VarID, c(\"CHR\", \"POS\", \"REF\", \"ALT\")))\n",
    "  assign(get_df, get(get_df) %>% mutate(refLen = nchar(get(get_df)$REF)))\n",
    "  assign(bed, data.frame(get(get_df)$CHR, get(get_df)$POS, as.numeric(get(get_df)$POS) + as.numeric(get(get_df)$refLen) - 1))\n",
    "\n",
    "  fwrite(get(bed), paste0(df_name, \"_non_sus_indels_bp\", i, \".bed\"), col.names=FALSE, sep = \"\\t\")\n",
    "}\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "0a9da533",
   "metadata": {},
   "source": [
    "Merge bedfile of gnomAD bins that are common and suspicious. \n",
    "Bash script below. "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "9ff935af",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'10'</li><li>'20'</li><li>'30'</li><li>'40'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '10'\n",
       "\\item '20'\n",
       "\\item '30'\n",
       "\\item '40'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '10'\n",
       "2. '20'\n",
       "3. '30'\n",
       "4. '40'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"10\" \"20\" \"30\" \"40\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system(\"bp_range=('10' '20' '30' '40')\n",
    "\n",
    "date=''\n",
    "output_path=''\n",
    "\n",
    "df_name='gnomAD'\n",
    "\n",
    "for i in ${bp_range[@]};\n",
    "do\n",
    "   echo $i\n",
    "   cat gnomad.exomes.r2.1.1.sites_indelsonly_rAF_bp${i}_SuspiciousIndels.lt50bp.bin.bed gnomad.exomes.r2.1.1.sites_indelsonly_rAF_bp${i}_CommonIndels.lt50bp.bin.bed > ${output_path}${df_name}_suspicious_and_common_bins_bp${i}.bed\n",
    "done;\", intern = TRUE)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "42d969bd",
   "metadata": {},
   "source": [
    "Bedtools intersect IGM indels that are non-suspicious by IGM indel with gnomAD bins that are common or suspicious. \n",
    "Bash script below."
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "5719ecf5",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<style>\n",
       ".list-inline {list-style: none; margin:0; padding: 0}\n",
       ".list-inline>li {display: inline-block}\n",
       ".list-inline>li:not(:last-child)::after {content: \"\\00b7\"; padding: 0 .5ex}\n",
       "</style>\n",
       "<ol class=list-inline><li>'10'</li><li>'20'</li><li>'30'</li><li>'40'</li></ol>\n"
      ],
      "text/latex": [
       "\\begin{enumerate*}\n",
       "\\item '10'\n",
       "\\item '20'\n",
       "\\item '30'\n",
       "\\item '40'\n",
       "\\end{enumerate*}\n"
      ],
      "text/markdown": [
       "1. '10'\n",
       "2. '20'\n",
       "3. '30'\n",
       "4. '40'\n",
       "\n",
       "\n"
      ],
      "text/plain": [
       "[1] \"10\" \"20\" \"30\" \"40\""
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "system(\"bp_range=('10' '20' '30' '40')\n",
    "\n",
    "date=''\n",
    "output_path=''\n",
    "\n",
    "df_1_name='gnomAD'\n",
    "df_2_name='IGM'\n",
    "\n",
    "for i in ${bp_range[@]};\n",
    "do\n",
    "   echo $i\n",
    "   bedtools intersect -wa -a ${output_path}${df_2_name}_non_sus_indels_bp${i}.bed -b ${output_path}${df_1_name}_suspicious_and_common_bins_bp${i}.bed | sort -k1,1n -k2,2n | uniq > ${output_path}${df_2_name}_non_sus_indels_annotated_bp${i}.bed\n",
    "done;\", intern = TRUE)\n"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "ed1401b1",
   "metadata": {},
   "source": [
    "Get files with: \n",
    " 1. File 1 (all_sus): [indels that are sus by IGM rAF] + [indels that are overlapping with gnomAD sus and common bins]\n",
    " 2. File 2 (reduced_sus): [indels that are non-sus by IGM rAF] - [indels that are overlapping with gnomAD sus and common bins]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "id": "0c0a051b",
   "metadata": {},
   "outputs": [],
   "source": [
    "library(tidyverse)\n",
    "library(data.table)\n",
    "library(readxl)\n",
    "library(dplyr)\n",
    "library(stringr)\n",
    "\n",
    "## This script is to get a file with all the suspicious IGM indels, including the indels that are non-suspicious by IGM rAF but are common/suspicious by gnomAD rAF \n",
    "\n",
    "bp_range = c(\"10\", \"20\", \"30\", \"40\")\n",
    "output_path = \"\"\n",
    "\n",
    "df_name = \"IGM\"\n",
    "\n",
    "###############################################################\n",
    "###############################################################\n",
    "\n",
    "## Read in IGM indels that are suspicious by IGM rAF only \n",
    "sus_bp10 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp10_SuspiciousIndels.lt50bp.csv\")\n",
    "sus_bp20 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp20_SuspiciousIndels.lt50bp.csv\")\n",
    "sus_bp30 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp30_SuspiciousIndels.lt50bp.csv\")\n",
    "sus_bp40 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp40_SuspiciousIndels.lt50bp.csv\")\n",
    "\n",
    "## Read in non-suspicious IGM indels \n",
    "nonsus_bp10 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp10_RareIndels.lt50bp.csv\")\n",
    "nonsus_bp20 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp20_RareIndels.lt50bp.csv\")\n",
    "nonsus_bp30 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp30_RareIndels.lt50bp.csv\")\n",
    "nonsus_bp40 = fread(\"2023-03-23_IGM_n39367_indelsonly_rAF_bp40_RareIndels.lt50bp.csv\")\n",
    "\n",
    "## Read in non-suspicous IGM indels bedfile that are annotated to be non-suspicous by gnomAD rAF\n",
    "annotated_bp10 = fread(\"IGM_non_sus_indels_annotated_bp10.bed\")\n",
    "annotated_bp20 = fread(\"IGM_non_sus_indels_annotated_bp20.bed\")\n",
    "annotated_bp30 = fread(\"IGM_non_sus_indels_annotated_bp30.bed\")\n",
    "annotated_bp40 = fread(\"IGM_non_sus_indels_annotated_bp40.bed\")\n",
    "\n",
    "###############################################################\n",
    "###############################################################\n",
    "\n",
    "## Map the annotated_bp indels to the nonsus_bp dataframes by CHR and first POS\n",
    "\n",
    "for (i in bp_range) {\n",
    "  nonsus_bp = paste0(\"nonsus_bp\", i)\n",
    "  annotated_bp = paste0(\"annotated_bp\", i)\n",
    "  \n",
    "  varID_annotated_bp = paste0(\"varID_annotated_bp\", i)\n",
    "  \n",
    "  assign(nonsus_bp, get(nonsus_bp) %>% separate(VarID, c(\"CHR\", \"POS\", \"REF\", \"ALT\")))\n",
    "  \n",
    "  assign(varID_annotated_bp, get(nonsus_bp) %>% filter(paste0(get(nonsus_bp)$CHR, \"-\", get(nonsus_bp)$POS) %in% paste0(get(annotated_bp)$V1, \"-\", get(annotated_bp)$V2)))\n",
    "}\n",
    "\n",
    "## Combine the suspicious indels that are suspicious by IGM rAF with the annotated non-suspicious indels that are suspicious by gnomAD rAF \n",
    "\n",
    "for (i in bp_range) {\n",
    "  varID_annotated_bp = paste0(\"varID_annotated_bp\", i)\n",
    "  assign(varID_annotated_bp, subset((get(varID_annotated_bp) %>% mutate(VarID = paste0(CHR, \"-\", POS, \"-\", REF, \"-\", ALT))), select = -c(CHR, POS, REF, ALT)))\n",
    "  \n",
    "  ## bind to suspicious indels by IGM rAF\n",
    "  sus_bp = paste0(\"sus_bp\", i)\n",
    "  \n",
    "  ## all suspicious (including annotated one)\n",
    "  all_sus_bp = paste0(\"all_sus_bp\", i)\n",
    "  \n",
    "  assign(all_sus_bp, rbind(get(varID_annotated_bp), get(sus_bp)))\n",
    "  fwrite(get(all_sus_bp), paste0(output_path, df_name, \"_all_suspicious_bp\", i, \".csv\"))\n",
    "}\n",
    "\n",
    "## Find the reduced non-suspicous indels.\n",
    "## Non-suspicious indels that are NOT in gnomAD bins that are common/sus. \n",
    "for (i in bp_range) {\n",
    "  nonsus_bp = paste0(\"nonsus_bp\", i)\n",
    "  assign(nonsus_bp, subset((get(nonsus_bp) %>% mutate(VarID = paste0(CHR, \"-\", POS, \"-\", REF, \"-\", ALT))), select = -c(CHR, POS, REF, ALT)))\n",
    "  \n",
    "  varID_annotated_bp = paste0(\"varID_annotated_bp\", i)\n",
    "  reduced_nonsus_bp = paste0(\"reduced_nonsus_bp\", i)\n",
    "  \n",
    "  # exclude VarIDs that are in varID_annotated_bp from the nonsus_bp df \n",
    "  assign(reduced_nonsus_bp, get(nonsus_bp) %>% filter(!(VarID %in% get(varID_annotated_bp)$VarID)))\n",
    "  fwrite(get(reduced_nonsus_bp), paste0(output_path, df_name, \"_reduced_nonsus_bp\", i, \".csv\"))\n",
    "  \n",
    "}"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "402c26af",
   "metadata": {},
   "source": [
    "Find the number of individuals that contain only suspicious deleterious indels by not counting the individuals that are also in the cohort with at least one suspicious deleterious indel in a constrained gene associated with an autosomal dominant disorder. \n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "85960a5a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<table class=\"dataframe\">\n",
       "<caption>A data.frame: 4 × 7</caption>\n",
       "<thead>\n",
       "\t<tr><th scope=col>bp_range</th><th scope=col>nonsus_individuals</th><th scope=col>only_sus_individuals</th><th scope=col>prct_nonsus_out_of_total</th><th scope=col>prct_only_sus_out_of_total</th><th scope=col>prct_only_sus_out_of_nonsus</th><th scope=col>prct_only_sus_out_of_5299</th></tr>\n",
       "\t<tr><th scope=col>&lt;chr&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;int&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th><th scope=col>&lt;dbl&gt;</th></tr>\n",
       "</thead>\n",
       "<tbody>\n",
       "\t<tr><td>10</td><td>4113</td><td>1186</td><td>10.45</td><td>3.01</td><td>28.84</td><td>22.38</td></tr>\n",
       "\t<tr><td>20</td><td>3834</td><td>1465</td><td> 9.74</td><td>3.72</td><td>38.21</td><td>27.65</td></tr>\n",
       "\t<tr><td>30</td><td>3617</td><td>1682</td><td> 9.19</td><td>4.27</td><td>46.50</td><td>31.74</td></tr>\n",
       "\t<tr><td>40</td><td>3412</td><td>1887</td><td> 8.67</td><td>4.79</td><td>55.30</td><td>35.61</td></tr>\n",
       "</tbody>\n",
       "</table>\n"
      ],
      "text/latex": [
       "A data.frame: 4 × 7\n",
       "\\begin{tabular}{lllllll}\n",
       " bp\\_range & nonsus\\_individuals & only\\_sus\\_individuals & prct\\_nonsus\\_out\\_of\\_total & prct\\_only\\_sus\\_out\\_of\\_total & prct\\_only\\_sus\\_out\\_of\\_nonsus & prct\\_only\\_sus\\_out\\_of\\_5299\\\\\n",
       " <chr> & <int> & <int> & <dbl> & <dbl> & <dbl> & <dbl>\\\\\n",
       "\\hline\n",
       "\t 10 & 4113 & 1186 & 10.45 & 3.01 & 28.84 & 22.38\\\\\n",
       "\t 20 & 3834 & 1465 &  9.74 & 3.72 & 38.21 & 27.65\\\\\n",
       "\t 30 & 3617 & 1682 &  9.19 & 4.27 & 46.50 & 31.74\\\\\n",
       "\t 40 & 3412 & 1887 &  8.67 & 4.79 & 55.30 & 35.61\\\\\n",
       "\\end{tabular}\n"
      ],
      "text/markdown": [
       "\n",
       "A data.frame: 4 × 7\n",
       "\n",
       "| bp_range &lt;chr&gt; | nonsus_individuals &lt;int&gt; | only_sus_individuals &lt;int&gt; | prct_nonsus_out_of_total &lt;dbl&gt; | prct_only_sus_out_of_total &lt;dbl&gt; | prct_only_sus_out_of_nonsus &lt;dbl&gt; | prct_only_sus_out_of_5299 &lt;dbl&gt; |\n",
       "|---|---|---|---|---|---|---|\n",
       "| 10 | 4113 | 1186 | 10.45 | 3.01 | 28.84 | 22.38 |\n",
       "| 20 | 3834 | 1465 |  9.74 | 3.72 | 38.21 | 27.65 |\n",
       "| 30 | 3617 | 1682 |  9.19 | 4.27 | 46.50 | 31.74 |\n",
       "| 40 | 3412 | 1887 |  8.67 | 4.79 | 55.30 | 35.61 |\n",
       "\n"
      ],
      "text/plain": [
       "  bp_range nonsus_individuals only_sus_individuals prct_nonsus_out_of_total\n",
       "1 10       4113               1186                 10.45                   \n",
       "2 20       3834               1465                  9.74                   \n",
       "3 30       3617               1682                  9.19                   \n",
       "4 40       3412               1887                  8.67                   \n",
       "  prct_only_sus_out_of_total prct_only_sus_out_of_nonsus\n",
       "1 3.01                       28.84                      \n",
       "2 3.72                       38.21                      \n",
       "3 4.27                       46.50                      \n",
       "4 4.79                       55.30                      \n",
       "  prct_only_sus_out_of_5299\n",
       "1 22.38                    \n",
       "2 27.65                    \n",
       "3 31.74                    \n",
       "4 35.61                    "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "## Table 3.\n",
    "\n",
    "library(tidyverse)\n",
    "library(data.table)\n",
    "library(readxl)\n",
    "library(dplyr)\n",
    "library(stringr)\n",
    "\n",
    "bp_range = c(\"10\", \"20\", \"30\", \"40\")\n",
    "\n",
    "total_samples = 39367\n",
    "\n",
    "variants_effects = c(\"frameshift_variant\", \"splice_donor_variant\", \"splice_acceptor_variant\", \"stop_gained\", \"start_lost\", \"stop_lost\", \"exon_loss_variant\")\n",
    "\n",
    "## This script finds the number of individuals with only deleterious suspicious indels in constrained, autosomal dominant genes  \n",
    "all_sus_bp10 = fread(\"IGM_all_suspicious_bp10.csv\")\n",
    "all_sus_bp20 = fread(\"IGM_all_suspicious_bp20.csv\")\n",
    "all_sus_bp30 = fread(\"IGM_all_suspicious_bp30.csv\")\n",
    "all_sus_bp40 = fread(\"IGM_all_suspicious_bp40.csv\")\n",
    "\n",
    "## Read in non-suspicious IGM indels (removed the indels that are in common/sus gnomAD bins)\n",
    "reduced_nonsus_bp10 = fread(\"IGM_reduced_nonsus_bp10.csv\")\n",
    "reduced_nonsus_bp20 = fread(\"IGM_reduced_nonsus_bp20.csv\")\n",
    "reduced_nonsus_bp30 = fread(\"IGM_reduced_nonsus_bp30.csv\")\n",
    "reduced_nonsus_bp40 = fread(\"IGM_reduced_nonsus_bp40.csv\")\n",
    "\n",
    "sample_name_gene_name = fread(\"2023-03-24_11-47-14_IGM_n39367_indels_genotypes_selectcols.csv\") \n",
    "colnames(sample_name_gene_name) = c(\"VarID\", \"geneName\", \"sampleName\", \"coveredCtrl\", \"AC\")\n",
    "\n",
    "\n",
    "effects = fread(\"2023-04-21_IGM_n39367_indels_genotypes_effects.csv\", header = TRUE)\n",
    "colnames(effects) = c(\"VarID\", \"Effect\")\n",
    "\n",
    "annotations = distinct(fread (\"2023-04-21_IGM_genename_gnomadpli_gnomadloeuf_omimdisease.csv\")) #18231 \n",
    "colnames(annotations) = c(\"geneName\", \"pLI\", \"oe_lof_upper\", \"OMIM_disease\")\n",
    "\n",
    "annotations[geneName == \"'HTT'\"]$OMIM_disease = \"Huntington disease, 143100 (3), Autosomal dominant\"\n",
    "annotations[geneName == \"'GLTSCR1'\"]$geneName = \"BICRA\"\n",
    "annotations[geneName == \"'FAM46A'\"]$geneName = \"TENT5A\"\n",
    "\n",
    "\n",
    "## Merge IGM all suspicious indels with gene name, sample name, pLI, loeuf, omim, effects\n",
    "\n",
    "for (i in bp_range) {\n",
    "  all_sus_bp = paste0(\"all_sus_bp\", i)\n",
    "  \n",
    "  sus_with_gene_bp = paste0(\"sus_with_gene_bp\", i)\n",
    "  sus_with_effects_bp = paste0(\"sus_with_effects_bp\", i)\n",
    "  sus_with_annot_bp = paste0(\"sus_with_annot_bp\", i)\n",
    "  \n",
    "  assign (sus_with_gene_bp, merge(get(all_sus_bp), sample_name_gene_name, by=\"VarID\" ))\n",
    "  assign (sus_with_effects_bp, merge(get(sus_with_gene_bp), effects, by=\"VarID\"))\n",
    "  assign (sus_with_annot_bp, merge(get(sus_with_effects_bp), annotations, by=\"geneName\"))\n",
    "  ## sus dataframe with all the annotations is sus_with_annot_bp\n",
    "  \n",
    "  ## Now only look for suspicious indels of interest by filtering for indels that are autosomal dominant, deleterious, and constrained (pLI > 0.5; oe_lof_upper < 0.35)\n",
    "  filtered_sus_bp = paste0(\"filtered_sus_bp\", i)\n",
    "  assign(filtered_sus_bp, get(sus_with_annot_bp)[OMIM_disease %like% \"Autosomal dominant\", ] %>% filter(Effect %in% variants_effects) %>% filter(oe_lof_upper < 0.35) %>% filter(pLI > 0.5))\n",
    "  \n",
    "  ### Now look for non-suspicious indels \n",
    "  \n",
    "  reduced_nonsus_bp = paste0(\"reduced_nonsus_bp\", i)\n",
    "  \n",
    "  nonsus_with_gene_bp = paste0(\"nonsus_with_gene_bp\", i)\n",
    "  nonsus_with_effects_bp = paste0(\"nonsus_with_effects_bp\", i)\n",
    "  nonsus_with_annot_bp = paste0(\"nonsus_with_annot_bp\", i)\n",
    "  ## reduced nonsus dataframe with all the annotations is nonsus_with_annot_bp\n",
    "  \n",
    "  assign(nonsus_with_gene_bp, merge(get(reduced_nonsus_bp), sample_name_gene_name, by = \"VarID\"))\n",
    "  assign(nonsus_with_effects_bp, merge(get(nonsus_with_gene_bp), effects, by = \"VarID\"))\n",
    "  assign(nonsus_with_annot_bp, merge(get(nonsus_with_effects_bp), annotations, by =\"geneName\"))\n",
    "  \n",
    "  ## Now only look for non suspicious indels of interest by filtering for indels that are autosomal dominant, deleterious, and constrained (pLI > 0.5; oe_lof_upper < 0.35)\n",
    "  filtered_nonsus_bp = paste0(\"filtered_nonsus_bp\", i)\n",
    "  assign(filtered_nonsus_bp, get(nonsus_with_annot_bp)[OMIM_disease %like% \"Autosomal dominant\", ] %>% filter(Effect %in% variants_effects) %>% filter(oe_lof_upper < 0.35) %>% filter(pLI > 0.5))\n",
    "  \n",
    "  ## find unique individuals that are in filtered_nonsus_bp \n",
    "  nonsus_individuals = paste0(\"only_sus_individuals_bp\", i)\n",
    "  num_nonsus_individuals= paste0(\"num_nonsus_individuals_bp\", i)\n",
    "  \n",
    "  assign(nonsus_individuals, as.data.frame(unique(get(filtered_nonsus_bp)$sampleName)))\n",
    "  assign(num_nonsus_individuals, nrow(get(nonsus_individuals)))\n",
    "  \n",
    "  ## find unique sus individuals that are in filtered_nonsus_bp \n",
    "  ## this means sus indels that are NOT in non_sus indels \n",
    "  only_sus_individuals = paste0(\"only_sus_individuals_bp\", i )\n",
    "  num_only_sus_individuals = paste0(\"num_only_sus_individuals_bp\", i)\n",
    "  \n",
    "  assign(only_sus_individuals, as.data.frame(unique((get(filtered_sus_bp) %>% filter(!(get(filtered_sus_bp)$sampleName %in% get(filtered_nonsus_bp)$sampleName)))$sampleName)) )\n",
    "  assign(num_only_sus_individuals, nrow(get(only_sus_individuals)))\n",
    "}\n",
    "\n",
    "## make dataframe to visualize data \n",
    "nonsus_individuals = c(num_nonsus_individuals_bp10, num_nonsus_individuals_bp20, num_nonsus_individuals_bp30, num_nonsus_individuals_bp40)\n",
    "only_sus_individuals = c(num_only_sus_individuals_bp10, num_only_sus_individuals_bp20, num_only_sus_individuals_bp30, num_only_sus_individuals_bp40)\n",
    "  \n",
    "summary = data.frame(bp_range, nonsus_individuals, only_sus_individuals)\n",
    "\n",
    "## Find proportion out of all IGM cohort \n",
    "summary$prct_nonsus_out_of_total = round((as.numeric(summary$nonsus_individuals) / total_samples) * 100, 2) \n",
    "summary$prct_only_sus_out_of_total = round((as.numeric(summary$only_sus_individuals) / total_samples) * 100, 2) \n",
    "summary$prct_only_sus_out_of_nonsus = round((as.numeric(summary$only_sus_individuals) / (as.numeric(summary$nonsus_individuals))) * 100, 2) \n",
    "summary$prct_only_sus_out_of_5299 = round((as.numeric(summary$only_sus_individuals) / (5299)) * 100, 2) \n",
    "\n",
    "\n",
    "summary "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "bb737478",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "R",
   "language": "R",
   "name": "ir"
  },
  "language_info": {
   "codemirror_mode": "r",
   "file_extension": ".r",
   "mimetype": "text/x-r-source",
   "name": "R",
   "pygments_lexer": "r",
   "version": "4.2.1"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
